dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,276 @@
+package com.dotmarketing.business.cache.provider;
+
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.github.benmanes.caffeine.base.UnsafeAccess;
+
+
+public class CacheSizingUtil {
+
+
+    final int standardSampleSize = Config.getIntProperty("CACHE_SIZER_SAMPLE_SIZE", 20);
+    
+    final int maxRecusionDepth = 10;
+    
+    final boolean useCompressedOops = true;
+    
+    public String averageSizePretty(final Map<String, Object> cacheMap) {
+
+        return UtilMethods.prettyByteify(averageSize(cacheMap));
+
+    }
+
+    /**
+     * Takes a map and pulls random values from it to calculate an average object size
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long averageSize(final Map<String, Object> cacheMap) {
+
+        if (cacheMap.size() == 0) {
+            return 0;
+        }
+
+        long totalSize = 0;
+
+        final int numberToSample = Math.min(cacheMap.size(), standardSampleSize);
+
+
+        for (int i = 0; i < numberToSample; i++) {
+            List<String> keysAsArray = new ArrayList<String>(cacheMap.keySet());
+            Random r = new Random();
+            Object object = cacheMap.get(keysAsArray.get(r.nextInt(keysAsArray.size())));
+            if (object instanceof Optional && ((Optional) object).isPresent()) {
+                object = ((Optional) object).get();
+            }
+            if (object instanceof Optional && !((Optional) object).isPresent()) {
+                return 0;
+            }
+            totalSize += sizeOf(object);
+        }
+        return totalSize / numberToSample;
+
+    }
+
+    /**
+     * Tries to use the Unsafe class to calculate an objects size. If this does not work, it will fall
+     * back on the serialzied size (which less than accurate).
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long sizeOf(Object object) {
+        long retainedSize = 0;
+        try {
+            retainedSize = retainedSize(object);
+        } catch (Throwable t) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)